pkg/cri/resource-manager/policy/builtin/topology-aware/resources.go

```diff
@@ -586,10 +617,14 @@ func (cs *supply) Allocate(r Request) (Grant, error) {
 			cs.node.Name(), full, cs.sharable, cs.AllocatableSharedCPU())
 	}
 
+	grant := newGrant(cs.node, cr.GetContainer(), cr.cpuType, exclusive, 0, 0, nil, 0)
+	grant.AccountAllocateCPU()
```

Comment:
I'm really not that familiar with the code but would it make the code simpler if we moved this after the fractional allocation below - would remove the need for the release calls there (?)

Reply:
It wouldn't make to code any simpler IMO. After a failure you'd still need to return the exclusively allocated CPUs to the to where they were carved out from (call ReleaseCPU(), because takeCPU() has already removed them). AccountAllocate was deliberately put where it is so that after that point we are in fully internally consistent state: exclusive part of the allocation both done and accounted for.